pyccel/ast/internals.py

```diff
@@ -163,3 +162,74 @@ def __str__(self):
             stop = str(self.stop)
         return '{0} : {1}'.format(start, stop)
 
+class Symbol(Basic):
+    """
+    Represent variable with undefined type
```

Comment:
This docstring needs a bit more explanation.Is the following accurate?
```
Symbolic placeholder for a Python variable, which has a name but no type yet.
This is very generic, and it can also represent a function or a module.
```

Reply:
done in 2b0aea0